Sources/PrimerSDK/Classes/User Interface/OAuth/OAuthViewController.swift

```diff
@@ -49,16 +49,21 @@ internal class OAuthViewController: PrimerViewController {
                     
                     let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
                     
+                    let routerDelegate: RouterDelegate = DependencyContainer.resolve()
+                    
+                    let router = routerDelegate as! Router
+                    
                     if settings.hasDisabledSuccessScreen {
-                        let routerDelegate: RouterDelegate = DependencyContainer.resolve()
-                        let router = routerDelegate as! Router
+                        
                         let rootViewController = router.root
                         
                         UIView.animate(withDuration: 0.3) {
                             (rootViewController?.presentationController as? PresentationController)?.blurEffectView.alpha = 0.0
                         } completion: { (_) in
                             rootViewController?.dismiss(animated: true, completion: nil)
                         }
+                    } else {
```

Comment:
I don't remember that clearly, but I think there's a reason I hadn't added this 'else' statement. Have you checked that it doesn't break the Klarna flow, cause OAuthViewController is re-used.

Reply:
Cheers yeah, I'll be performing some more QA, especially for Klarna (just wanted to start the discussion here). Still, it's probably best if we enable this UI for PayPal at least. MIght refactor and do this only for PayPal.